pandas/core/arrays/string_.py

```diff
@@ -298,6 +298,13 @@ def astype(self, dtype, copy=True):
             arr[mask] = 0
             values = arr.astype(dtype.numpy_dtype)
             return IntegerArray(values, mask, copy=False)
+        elif np.issubdtype(dtype, np.floating):
```

Comment:
can this not also be a _FloatingDtype?

Reply:
Note that
```python
import numpy as np
from pandas.core.dtypes.common import pandas_dtype
from pandas.core.arrays.floating import FloatingDtype

isinstance(pandas_dtype("float"), FloatingDtype)  # False

np.issubdtype(dtype, np.floating)
E       TypeError: data type not understood
```
I added some code that handles the `FloatingDtype` explicitly.